docs/dev/framework/security/data_container.md

```diff
@@ -0,0 +1,93 @@
+---
+title: "Data Container permissions"
+description: Explanation of creating data container voters.
+aliases:
+    - /framework/security/data_container/
+---
+
+Since Contao 5 data container permissions can be set by data container voters. This replaces the "checkPermission" onload callbacks that were used in Contao 4 and before.
+
+Contao provides an abstract class [AbstractDataContainerVoter](https://github.com/contao/contao/blob/5.3/core-bundle/src/Security/Voter/DataContainer/AbstractDataContainerVoter.php) for data container voters.
+A class extending it is able to vote for data container actions. See the following example:
+
+```php
+namespace App\Security\Voter\DataContainer;
+
+use App\Model\ExampleArchiveModel;
+use App\Security\ExamplePermissions;
+use Contao\CoreBundle\Security\DataContainer\CreateAction;
+use Contao\CoreBundle\Security\DataContainer\DeleteAction;
+use Contao\CoreBundle\Security\DataContainer\ReadAction;
+use Contao\CoreBundle\Security\DataContainer\UpdateAction;
+use Contao\CoreBundle\Security\Voter\DataContainer\AbstractDataContainerVoter;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\AccessDecisionManagerInterface;
+
+class ExampleAccessVoter extends AbstractDataContainerVoter
+{
+    public function __construct(
+        private readonly AccessDecisionManagerInterface $accessDecisionManager
+    ) {
+    }
+
+    protected function getTable(): string
+    {
+        return ExampleArchiveModel::getTable();
```

Comment:
I would suggest to use a string here, because models are not necessary and usually not used in the back end. It also does not exist in the code example and might confuse readers.